usr/src/lib/libzfs/common/libzfs_dataset.c

```diff
@@ -3628,8 +3629,6 @@ int
 zfs_promote(zfs_handle_t *zhp)
 {
 	libzfs_handle_t *hdl = zhp->zfs_hdl;
-	zfs_cmd_t zc = { 0 };
-	char parent[MAXPATHLEN];
```

Comment:
I think it's just left over code.  I can't think why else you'd do that string copy.